frontend/packages/console-shared/src/hooks/__tests__/perspective-utils.spec.ts

```diff
@@ -0,0 +1,252 @@
+        },
+      },
+    ]);
+  });
+
+  it('should return all the available perspectives if perspectives are not set in the server flags', async () => {
+    window.SERVER_FLAGS.perspectives = undefined;
+
+    const { result } = testHook(() => usePerspectives());
+
+    expect(result.current).toEqual([
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'admin',
+          name: 'Administrator',
+        },
+      },
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'dev',
+          name: 'Developer',
+          defaultPins: [{ kind: 'ConfigMap' }, { kind: 'Secret' }],
+        },
+      },
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'dev-test',
+          name: 'Test Developer',
+          defaultPins: [{ kind: 'Deployments' }, { kind: 'Secret' }],
+        },
+      },
+    ]);
+  });
+
+  it('should return all the available perspectives if perspectives are not configured in the server flags', async () => {
+    window.SERVER_FLAGS.perspectives = '';
+
+    const { result } = testHook(() => usePerspectives());
+
+    expect(result.current).toEqual([
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'admin',
+          name: 'Administrator',
+        },
+      },
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'dev',
+          name: 'Developer',
+          defaultPins: [{ kind: 'ConfigMap' }, { kind: 'Secret' }],
+        },
+      },
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'dev-test',
+          name: 'Test Developer',
+          defaultPins: [{ kind: 'Deployments' }, { kind: 'Secret' }],
+        },
+      },
+    ]);
+  });
+
+  it('should return only the enabled perspectives and the perspectives that satisfy the missing accessreview checks that are set in the server flags', async () => {
+    window.SERVER_FLAGS.perspectives =
+      '[{ "id" : "dev", "visibility": {"state": "AccessReview" , "accessReview": { "missing" : [{ "resource" : "namespaces", "verb" : "list" }]}}}, { "id" : "dev-test", "visibility": {"state" : "Enabled" }}, { "id" : "admin", "visibility": {"state" : "Disabled" }}]';
+
+    (checkAccess as jest.Mock).mockReturnValue(Promise.resolve({ status: { allowed: true } }));
+    const { result, rerender } = testHook(() => usePerspectives());
+
+    await act(async () => {
+      rerender();
+    });
+    expect(result.current).toEqual([
+      {
+        type: 'Perspective',
+        properties: {
+          id: 'dev-test',
+          name: 'Test Developer',
+          defaultPins: [{ kind: 'Deployments' }, { kind: 'Secret' }],
+        },
+      },
+    ]);
+  });
+
+  it('should return the admin perspective as default if all the perspectives are disabled or accessreview checks fail', async () => {
+    (checkAccess as jest.Mock).mockReturnValue(Promise.resolve({ status: { allowed: true } }));
```

Comment:
Description "accessreview checks fail" doesn't match the `checkAccess` return value which is `{ allowed: true }`

One of both needs to reflect the other, or?